impl/src/main/java/org/pentaho/platform/genericfile/model/BaseGenericFile.java

```diff
@@ -34,6 +36,10 @@ public class BaseGenericFile implements IGenericFile {
   private String title;
   private String description;
   private String owner;
+  private Date createdDate;
+  private String creatorId;
+  private long fileSize;
+  private Map<String, Object> customProperties = new HashMap<>();
```

Comment:
Imo, and as far as I remember what we discussed, custom properties should not be part of IGenericFile. Instead, there should be a separate `IGenericFileMetadata` object, and corresponding `getFileMetadata` and `setFileMetadata` methods in the service.